lib/rust/parser/tests/parse.rs

```diff
@@ -0,0 +1,236 @@
+//! Parse expressions and compare their results to expected values.
+
+// === Non-Standard Linter Configuration ===
+#![allow(clippy::option_map_unit_fn)]
+#![allow(clippy::precedence)]
+#![allow(dead_code)]
+#![deny(non_ascii_idents)]
+#![deny(unconditional_recursion)]
+#![warn(unsafe_code)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+use lexpr::sexp;
+
+
+
+// ===========================
+// === Test support macros ===
+// ===========================
+
+/// Parses input as a sequence of S-expressions, and wraps it in a `Block`.
+macro_rules! block {
+    ( $statements:tt ) => {
+        sexp![(Block #($statements))]
+    }
+}
+
+
+
+// =============
+// === Tests ===
+// =============
+
+#[test]
+fn application() {
+    test("a b c", block![(App (App (Ident a) (Ident b)) (Ident c))]);
+}
+
+#[test]
+#[ignore]
+fn type_definition_bool() {
+    let bool = [
+        "type Bool",
+        "    False",
+        "    True"
+    ].join("\n");
+    let bool_ast = block![()];
+    test(&bool, bool_ast);
+}
+
+#[test]
+#[ignore]
+fn type_definition_geo() {
+    let geo = [
+        "type Geo",
+        "    Circle",
+        "    radius : Float",
+        "    Rectangle",
+        "    width  : Float",
+        "    height : Float",
+    ].join("\n");
+    let geo_ast = block![()];
+    test(&geo, geo_ast);
+}
+
+#[test]
+#[ignore]
+fn type_definition_option() {
+    let option = [
+        "type Option a",
+        "    Some elem:a",
+        "    None",
+    ].join("\n");
+    let option_ast = block![()];
+    test(&option, option_ast);
+}
```

Comment:
these tests are not enabled and are not finished. What is the plan for them?

Reply:
I shouldn't have included these. While I was writing tests to catch the `TypeDef` printing bug, I refactored the test runner API--I wanted to get the new interface in here since there'd be no point in reviewing the old one I was about to replace, but I pulled in the WIP test cases with it.